torch/csrc/autograd/variable.cpp

```diff
@@ -341,7 +341,6 @@ const std::shared_ptr<torch::autograd::Node>& VariableHooks::grad_fn(const Tenso
     if (diff_view_meta->attr_version != current_version) {
       // This is an indirect rebase_history due to another view or the base being modified inplace
       handle_view_on_rebase(diff_view_meta, /* indirect */ true);
-      TORCH_INTERNAL_ASSERT(diff_view_meta->output_nr_ == 0);
```

Comment:
We don't want to do that I think. Many other things are going to break if you do that.
I think what you want to do here is:
- In this function, change the original condition to only do the rebase if it is a view and not MULTI_OUTPUT_SAFE.
- In the rebase_history function, similarly, only do the view handling if it is not MULTI_OUTPUT_SAFE.

This will ensure that for now, these function have the exact same behavior as before (as if they were not views + no version tracking) and we only add the warning.
Next version, we can remove this CreationMeta and add back the version tracking.